# Question 17
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.8925)
loc_2 = z.locate(0.0495)
print(loc_1)
print(loc_2)

1.2399499999416776
-1.6497100000603182


$$5.205 - 1.24\sigma = \mu$$

$$1.65\sigma + 5.195 = \mu$$

$$5.205 - 1.24\sigma = 1.65\sigma + 5.195$$

$$2.9\sigma = 0.01$$

$$\sigma = 0.0034$$

$$\mu = (1.65 * 0.0034) + 5.195$$

$$\mu = 5.20061$$

In [3]:
dist = normal(5.20061, 0.0034)
undersize = dist.integrate(0, 5.195)
oversize = dist.integrate(5.205, 10)
satisfactory = dist.integrate(5.195, 5.205)
prob = undersize * oversize * satisfactory * math.factorial(3)
print("prob = " + str(prob))

prob = 0.024871310254222588
